paper.md

```diff
@@ -0,0 +1,159 @@
+
+
+# Code health processes
+
+There exists a wide variety of literature on best practices for developing healthy code (see for example [@martin2008] or [@perkel2022]). In this section we discuss several such practices that benefit from the existence of automated tooling. This list is not meant to be exhaustive, but each item should be considered important for developing a trustworthy and impactful software package.
+
+## Automated testing
+
+Automated testing is critical to ensuring the correctness of code during initial implementation and through ongoing changes. Code should be validated by a comprehensive suite of tests including unit tests, which confirm the accuracy of individual functions, and integration tests that ensure end-to-end functionality for supported use cases. A comprehensive and automated test suite is an indicator of a trustworthy software package.
+
+The LF-PPT supports automated testing by configuring several different continuous integration GitHub workflows, that run test suites automatically:
+ - at each push or pull request to ensure that changes to the current branch do not break the program’s behavior and
+ - in regularly scheduled smoke tests (usually daily) to ensure that the code has not broken due to a change of the project’s code or the behavior of its dependencies.
+
+## Documentation
+
+Thorough documentation of code allows others to use the package more readily and encourages maintainability. The LF-PPT supports the use of sphinx[^1] for integration with ReadTheDocs[^2] to automatically render Python docstrings into well formatted documentation alongside manually written documentation. Additionally, to demonstrate the use of the package in context, the LF-PPT provides automatic rendering of example Jupyter notebooks within the documentation.
+
+## Distribution
+
+Software is only useful to the broader community if other users can find, install, and update it. Package management systems such as pip[^3] or conda[^4] provide simple tools for a user to download a new software package and its dependencies from public repositories such as PyPI[^5] or conda-forge[^6]. To enable easy code installation with pip or conda the LF-PPT provides support for automatic distribution through both PyPI and conda-forge when the user applies a new git tag via the GitHub UI.
+
+## Additional code health tools
+
+The LF-PPT supports many more features than those listed here including static code analysis, performance benchmarking, code testing coverage and more. For a complete list, please see the documentation at http://lincc-ppt.readthedocs.io/.
+
+The most notable exclusion from the LF-PPT is code. Aside from a few optional stub source and test files, the LF-PPT allows the user to focus on scientific code, while supporting them with the industry best practices in maintainable software engineering.
```

Comment:
I found this paragraph a bit hard to understand (what is it trying to communicate and how does it relate to the "Additional code health tools" heading).  Can you elaborate on what you are trying to convey and how it relates to the rest of the subsection?   I mean we could maybe chat about it here in a comment in hopes of identifying potential edits to the text.

Reply:
Sure, good call here. What I want to convey is that this template provides the foundational pieces of a software package, but leaves the code (i.e. the science) to the user to implement. We want the template to lower the barrier to create a quality software package, without actually providing code.

I agree though that this can seem out of place. We don't call out that we provide code anywhere else, so I would be ok just cutting this paragraph. 

What are your thoughts?

Reply:
I think it would be fine to cut the paragraph.

Reply:
Hmm.  Thanks, so I think this reinforces my feeling that it doesn't belong here.  But I feel the idea that we are trying to make it easier for scientists to focus just on the code and not the infrastructure is a nice one and perhaps it could be worked (more concisely) into the summary at the top?

Reply:
Ok, I've cut this paragraph, and added a new line at the end of the summary paragraph:

"Notably the template does not include any application specific code. This allows the user to focus on their scientific code, and not building or maintaining code health infrastructure."